ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -358,31 +363,75 @@ def task_timer_thread(self):
         log = self.get_logger()
         loops = self._task.machine.states[self._task.state].timer_length
 
+        # Record the current state
+        curr_state = self._task.state
+
         # Publish a task update message once per second
         for i in range(int(loops)):
             self.publish_task_state_message()
-            self._task.machine.states[self._task.state].timer_length -= 1
+
+            try:
+                self._task.machine.states[self._task.state].timer_length -= 1
```

Comment:
Should this also use the new `curr_state` above? Otherwise, we may be modifying a different task's timing component on accident.

Reply:
Ahh, yeah good point. That is why I added the `try`. During testing `self._task` changed and threw an error.

Reply:
Oh, I see. Hadn't thought of that. I switched it to use `curr_state`.